### ISSUE TITLE
Histogram only shows results from the current write-active index

### ISSUE THREAD

**Issue by g60ocR** [Graylog2/graylog2-web-interface#1215]
When searching several days back, the histogram will only show results for the current write-active index. The search results often goes way more back then the histogram shows. Version 1.0.0 (VMWare OVA)

**Comment by g60ocR** [comment]
Am I the only one seeing this? This ruins dashboards and search results visually when looking at data more than 6-8 hours back in time.

**Comment by edmundoa** [comment]
Hi g60ocR,

My first guess is that your retention strategy is deleting or closing ES indices "too soon", making it impossible to store data for more than a few hours. Could you take a look at the list of indices shown on `System -> Indices` and verify that they contain messages older than those shown in the histogram?

**Comment by edmundoa** [comment]
By the way, it's usually better to ask questions on our [mailing list](https://groups.google.com/forum/#!forum/graylog2) or our IRC channel [#graylog on Freenode](https://webchat.freenode.net/?channels=%23graylog), as more people can help, and we keep GitHub to post bugs and/or features. :)

**Comment by g60ocR** [comment]
Your current configuration is 1,073,741,824 bytes per index and a maximum number of 10 indices.
1 indices with a total of 112,915,703 messages under management. Current write-active index is graylog_54.

Ok, I suspected this was a bug so I went straight to the issue tracker. It seems I only have one indice so far?

**Comment by kroepke** [comment]
Please note that the size based rotation is not overly accurate, because Elasticsearch's on disk representation means that the byte value will fluctuate a lot (due to merging parts of the index).

It's best to avoid it, because it will most certainly not do what you intended.

**Comment by g60ocR** [comment]
OK. I will change the retention strategy to time instead of size. I will keep you updated with the results.

**Comment by g60ocR** [comment]
I still have the same problem. I changed my retention to 1 index per day and 30 indexes. I have upgraded to 1.0.1 as well.

Your current configuration rotates the indices every 24 hours and keeps a maximum number of 30 indices.

1 indices with a total of 155,396,679 messages under management. Current write-active index is graylog_61.

I can't see any closed indexes under System -> Indices.

**Comment by edmundoa** [comment]
By default Graylog will delete your old indices, you can change that by setting `retention_strategy` to `close` in your Graylog server configuration.

Reading your message I'm not entirely sure rotation is working for you. When you go to System -> Indices, have you more than one index in there? If not, you may try to manually rotate the deflector once (look in the Action button on System -> Indices) and then wait to see if the rotation is actually working with the settings you defined.

**Comment by g60ocR** [comment]
I tried cycling the deflector. I get an 500 Server error status. I get this quite a lot by the way, when saving new extractors for example. But when I save extractors, I get 500 but the extractor is created/changed as it should. 

(You caused a org.graylog2.restclient.lib.APIException. API call failed POST http://@123.123.123.123:12900/system/deflector/cycle returned 500 Internal Server Error body: {"type":"ApiError","message":"{ \"serverUsed\" : \"127.0.0.1:27017\" , \"ok\" : 1 , \"n\" : 0 , \"updatedExisting\" : false , \"err\" : \"bad offset:0 accessing file: /var/opt/graylog/data/mongodb/graylog.0. See http://dochub.mongodb.org/core/data-recovery\" , \"code\" : 13440}"})

root@graylog:/var/opt/graylog/data/mongodb# ls -latr
total 163856
drwxr-xr-x 7 graylog graylog     4096 Apr 10 11:47 ..
drwxr-x--- 3 graylog graylog     4096 Apr 10 12:53 .
-rwx------ 1 graylog graylog        6 Apr 28 07:35 mongod.lock
-rw------- 1 graylog graylog 16777216 Apr 28 07:35 local.ns
-rw------- 1 graylog graylog 67108864 Apr 28 07:35 local.0
drwx------ 2 graylog graylog     4096 May  5 21:18 journal
-rw------- 1 graylog graylog 16777216 May  7 08:08 graylog.ns
-rw------- 1 graylog graylog 67108864 May  7 08:13 graylog.0
root@graylog:/var/opt/graylog/data/mongodb#

**Comment by joschi** [comment]
@g60ocR Looks like your MongoDB database has been at least in parts corrupted.

Did you follow the steps described in http://docs.mongodb.org/manual/tutorial/recover-data-following-unexpected-shutdown/ to repair the database?

**Comment by g60ocR** [comment]
Nope I didn't follow them, because didn't know the database was corrupt. :)

I ran the following command:
/opt/graylog/mongodb/bin/mongod --dbpath /var/opt/graylog/data/mongodb --repair

Didn't see anything strange in the output. Though mongod didn't want to come up when I started it using the "graylog-ctl start" command. It was a permission problem since I had run the repair using the root account. So I changed ownership of the files to graylog and then mongod woke up again.

Now everything is working as it should! Thank you so much.
